### DISCUSSION TITLE
404 page and redirect with vercel.json

### DISCUSSION THREAD

**Discussion by Crispy-Cream** [vercel/community#483] [Help] [answered]
Hey,

At the moment I am trying to figure out how I can setup a redirect from `/github` to `https://github.com/Crispy-Cream` while having a 404 page setup. 

My current vercel.json looks like this
```json
{
  "redirects": [
    {
      "source": "/github",
      "destination": "https://github.com/Crispy-Cream",
      "permanent": true
    }
  ]
}
```
Which works great for redirecting. the first thing I tried was
```json
{
   "routes": [
    { "handle": "filesystem" },
    { "src": "/(.*)", "status": 404, "dest": "/404.html" },
    { "src": "/github", "headers": { "Location": "https://github.com/Crispy-Cream" }, "status": 307 }
  ]
```
which wasn't working for the redirect or 404 page. my next try was having both a `routes` and a `redirects` but that throws a vercel error because you can't have them both in the json. 

What can I do to keep my redirect while making use of my 404 page using the vercel.json file?

thank you

**Comment by samcx** [comment, marked as answer]
Have you tried doing this?
```json
{
  "routes": [
    { "handle": "filesystem" },
    { "src": "/github", "status": 307, "dest": "https://github.com/Crispy-Cream" },
    { "src": "/(.*)", "status": 404, "dest": "/404.html" },
  ]
}
 ```
